h1b/src/gpio.rs

```diff
@@ -135,30 +135,79 @@ impl GPIOPin {
         }
 
         self.client.get().map(|client| {
-            client.fired(self.client_data.get());
+            client.fired(self.client_data.get())
         });
     }
+}
 
-    pub fn set_client(&self, client: &'static hil::gpio::Client) {
-        self.client.set(Some(client));
+impl hil::gpio::Configure for GPIOPin {
+    fn configuration(&self) -> hil::gpio::Configuration {
+        if self.is_output() {
+            hil::gpio::Configuration::InputOutput
+        } else {
+            hil::gpio::Configuration::Input
+        }
     }
-}
 
-impl hil::gpio::Pin for GPIOPin {
-    fn make_output(&self) {
+    fn make_output(&self) -> hil::gpio::Configuration {
         let port: &mut PortRegisters = unsafe { transmute(self.port) };
         port.output_enable.set(1 << (self.pin as u32));
+        hil::gpio::Configuration::InputOutput
     }
 
-    fn make_input(&self) {
+    fn disable_output(&self) -> hil::gpio::Configuration {
+        let port: &mut PortRegisters = unsafe { transmute(self.port) };
+        port.output_enable.set(1 << (self.pin as u32));
+        hil::gpio::Configuration::Input
+    }
+
+    fn make_input(&self) -> hil::gpio::Configuration {
         // Noop, input is always enabled on this chip
+        self.configuration()
     }
 
-    fn disable(&self) {
+    fn disable_input(&self) -> hil::gpio::Configuration {
+        // Noop, input is always enabled on this chip
+        self.configuration()
+    }
+
+    fn deactivate_to_low_power(&self) {
+        // TODO: I don't know what the lowest-power state of the pin is; this is
+        // just a guess. We may want to reconfigure pinmux here too, the same as
+        // we do to set a floating state.
```

Comment:
Leaving the pin floating is important for low power. Otherwise depending on if it's tied to ground or a voltage you can draw significant power.

Reply:
It now calls disable_output() and sets the pin floating.